File: easy/swap_case.hs
import System.Environment (getArgs)
import Data.Char (isLower, isUpper, toLower, toUpper)

swapCase  :: Char -> Char
swapCase c | isLower c = toUpper c
           | isUpper c = toLower c
           | otherwise = c

main :: IO ()
main = do
    [inpFile] <- getArgs
    input <- readFile inpFile
    putStr . unlines . map (map swapCase) $ lines input

File: easy/swap_case.scala
object Main extends App {
  val source = scala.io.Source.fromFile(args(0))
  val lines = source.getLines.filter(_.length > 0)
  def swp(c: Char): Char =
    if (c.isLower) c.toUpper else c.toLower

  for (l <- lines)
    println(l.map(swp))
}
